### PR TITLE
Cc artemnoskov

### PR DESCRIPTION
**Pull request by artemnoskov**
Pulling result of 5 sprints into main

### CODE REVIEW

File: src/main/java/com/github/hey_world_team/currency_converter/controller/rest/CentralBankApiController.java [hunk 1]
```diff
@@ -1,7 +1,42 @@
...
+    private final FileService fileService;
+
+    @Autowired
+    public CentralBankApiController(FileService fileService) {
+        this.fileService = fileService;
+    }
+
+    //TODO for future api with current date
+    //private static final String DATE_API = "date_req";
+    private static final String LINK = "http://www.cbr.ru/scripts/XML_daily.asp";
```
Comment (dmitr2ish) [inline src/main/java/com/github/hey_world_team/currency_converter/controller/rest/CentralBankApiController.java:26]: the link must be placed in properties and received via @Valuethe link must be placed in properties and received via @Value

File: src/main/java/com/github/hey_world_team/currency_converter/controller/rest/CentralBankApiController.java [hunk 2]
```diff
@@ -1,7 +1,42 @@
...
+        String answer = fileService.writeToFile(currenciesXml);
+        return new ResponseEntity<>("File was " + answer.toLowerCase(), HttpStatus.OK);
+    }
+
+    //Для тестирования методов
+    @GetMapping(value = "/test")
+    public ResponseEntity<String> createObject() throws IOException {
+        String answer = fileService.parseXmlToObject();
+        return new ResponseEntity<>("File was " + answer.toLowerCase(), HttpStatus.OK);
+    }
```
Comment (dmitr2ish) [inline src/main/java/com/github/hey_world_team/currency_converter/controller/rest/CentralBankApiController.java:41]: This method Is not need in main branch

File: src/main/java/com/github/hey_world_team/currency_converter/repository/CurrencyDataRepository.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Repository
+public class CurrencyDataRepository {
+    private Map<String, Double> repository = new HashMap<>();
+
+    public void save(String name, Double value) {
+
```
Comment (dmitr2ish) [inline src/main/java/com/github/hey_world_team/currency_converter/repository/CurrencyDataRepository.java:13]: excess line

File: src/main/java/com/github/hey_world_team/currency_converter/repository/CurrencyDataRepository.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        repository.put(name, value);
+    }
+
+    public Double getCurrencyValueByName(String name) {
+
```
Comment (dmitr2ish) [inline src/main/java/com/github/hey_world_team/currency_converter/repository/CurrencyDataRepository.java:18]: excess line

File: src/main/java/com/github/hey_world_team/currency_converter/repository/CurrencyDataRepository.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        Double value = repository.get(name);
+        return value;
+    }
+
+    public Map getAllCurrencies() {
+
```
Comment (dmitr2ish) [inline src/main/java/com/github/hey_world_team/currency_converter/repository/CurrencyDataRepository.java:24]: excess line

File: src/main/java/com/github/hey_world_team/currency_converter/repository/CurrencyDataRepository.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+        return repository;
+    }
+
+    public void updateCurrencyValueByName(String name, Double value) {
+
```
Comment (dmitr2ish) [inline src/main/java/com/github/hey_world_team/currency_converter/repository/CurrencyDataRepository.java:29]: excess line

File: src/main/java/com/github/hey_world_team/currency_converter/repository/CurrencyDataRepository.java [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+        repository.replace(name, value);
+    }
+
+    public void updateCurrencyValues() {
+
+        repository.replaceAll((name, value) -> getCurrencyValueByName(name));
+    }
```
Comment (dmitr2ish) [inline src/main/java/com/github/hey_world_team/currency_converter/repository/CurrencyDataRepository.java:36]: This method isnt working

File: src/main/java/com/github/hey_world_team/currency_converter/service/FileService.java [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+        logger.info("Write {} completed", FILE_FOREIGN_CURRENCIES);
+        return FileWriteStatus.WRITTEN.name();
+    }
+
+    public String parseXmlToObject() throws IOException {
+        logger.info("Started writing XML to object");
+        File input = new File(propertiesForFileService.getPath() + FILE_FOREIGN_CURRENCIES);
+        Document doc = Jsoup.parse(input, "windows-1251", "", Parser.xmlParser());
+
+        for (Element e : doc.select("Valute")) {
```
Comment (dmitr2ish) [inline src/main/java/com/github/hey_world_team/currency_converter/service/FileService.java:53]: Valute - must be constant

File: src/main/java/com/github/hey_world_team/currency_converter/service/FileService.java [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+            return FileWriteStatus.NOT_WRITTEN.name();
+        }
+        logger.info("Write {} completed", FILE_FOREIGN_CURRENCIES);
+        return FileWriteStatus.WRITTEN.name();
+    }
+
+    public String parseXmlToObject() throws IOException {
+        logger.info("Started writing XML to object");
+        File input = new File(propertiesForFileService.getPath() + FILE_FOREIGN_CURRENCIES);
+        Document doc = Jsoup.parse(input, "windows-1251", "", Parser.xmlParser());
```
Comment (dmitr2ish) [inline src/main/java/com/github/hey_world_team/currency_converter/service/FileService.java:51]: charset - must be constant

File: src/test/java/com/github/hey_world_team/currency_converter/repository/CurrencyDataRepositoryTest.java [hunk 10]
```diff
@@ -0,0 +1,31 @@
+package com.github.hey_world_team.currency_converter.repository;
+
+import static org.junit.Assert.*;
+
+public class CurrencyDataRepositoryTest {
```
Comment (dmitr2ish) [inline src/test/java/com/github/hey_world_team/currency_converter/repository/CurrencyDataRepositoryTest.java:5]: It's just an empty class, why is it in the main branch

File: src/main/java/com/github/hey_world_team/currency_converter/controller/rest/CentralBankApiController.java (hunk 1 above)
Comment (artemnoskov) [reply src/main/java/com/github/hey_world_team/currency_converter/controller/rest/CentralBankApiController.java]: Done!

File: src/main/java/com/github/hey_world_team/currency_converter/controller/rest/CentralBankApiController.java (hunk 2 above)
Comment (artemnoskov) [reply src/main/java/com/github/hey_world_team/currency_converter/controller/rest/CentralBankApiController.java]: Done!

File: src/main/java/com/github/hey_world_team/currency_converter/repository/CurrencyDataRepository.java (hunk 4 above)
Comment (artemnoskov) [reply src/main/java/com/github/hey_world_team/currency_converter/repository/CurrencyDataRepository.java]: Done!

File: src/main/java/com/github/hey_world_team/currency_converter/repository/CurrencyDataRepository.java (hunk 5 above)
Comment (artemnoskov) [reply src/main/java/com/github/hey_world_team/currency_converter/repository/CurrencyDataRepository.java]: Done!

File: src/main/java/com/github/hey_world_team/currency_converter/repository/CurrencyDataRepository.java (hunk 6 above)
Comment (artemnoskov) [reply src/main/java/com/github/hey_world_team/currency_converter/repository/CurrencyDataRepository.java]: Done!

File: src/main/java/com/github/hey_world_team/currency_converter/repository/CurrencyDataRepository.java (hunk 7 above)
Comment (artemnoskov) [reply src/main/java/com/github/hey_world_team/currency_converter/repository/CurrencyDataRepository.java:31]: Done!

File: src/main/java/com/github/hey_world_team/currency_converter/service/FileService.java (hunk 8 above)
Comment (artemnoskov) [reply src/main/java/com/github/hey_world_team/currency_converter/service/FileService.java]: Done!

File: src/main/java/com/github/hey_world_team/currency_converter/service/FileService.java (hunk 9 above)
Comment (artemnoskov) [reply src/main/java/com/github/hey_world_team/currency_converter/service/FileService.java]: Done!

File: src/test/java/com/github/hey_world_team/currency_converter/repository/CurrencyDataRepositoryTest.java (hunk 10 above)
Comment (artemnoskov) [reply src/test/java/com/github/hey_world_team/currency_converter/repository/CurrencyDataRepositoryTest.java:10]: Done!

**Pull request merged**